src/scrape_up/steam/steamScraper.py

```diff
@@ -0,0 +1,206 @@
+            doc = BeautifulSoup(response.content, "html.parser")
+            games = doc.find_all("div", {"class": "responsive_search_name_combined"})
+
+            for game in games:
+                game_info = self._extract_game_info(game)
+                all_game_info.append([*game_info, filter])
+
+                if len(all_game_info) >= n0Games:
+                    break
+            if len(all_game_info) >= n0Games:
+                break
+        return all_game_info
+
+    def Scraper(self, n0Games=5, tags=["Discounts", "F2P"]):
+        """
+        Scrapes game data for each tag specified in the 'tags' attribute and returns a dictionary.
+
+        Returns:
+            dict: Dictionary containing game data.
+
+        Tags:
+            The 'tags' attribute specifies the filters to be applied during scraping.
+            These tags correspond to keys in the 'search_filters' dictionary, which maps each tag to its respective filter query parameter.
+
+            Example:
+            If 'tags' is ['Discounts', 'F2P'], the scraper will retrieve games that are discounted and free to play.
+            The available tags are:
+                Top Sellers, Discounts, F2P (Free to Play)
+                Software, DLC, Demos, Mods, Games
+                Windows, MacOS, Linux, VR, Deck
+                MP, PvP, Co-op
+        """
+        all_data = []
+        search_filters = {
+            "Top Sellers": "topselling",
+            "Discounts": "specials=1",
+            "F2P": "?maxprice=free",
+            "Software": "category1=994",
+            "DLC": "category1=21",
+            "Demos": "category1=10",
+            "Mods": "category1=997",
+            "Games": "category1=998",
+            "Windows": "os=win",
+            "MacOS": "os=mac",
+            "Linux": "os=linux",
+            "VR": "vrsupport=402",
+            "Deck": "deck_compatibility=2",
+            "MP": "category3=1",
+            "PvP": "category3=49",
+            "Co-op": "category3=9",
+        }
+        for filter in tags:
+            url = f"{self.base_url}{search_filters[filter]}"
+            filter_data = self._scrape_page(url, filter, n0Games)
+            all_data.extend(filter_data)
+
+        data = {col: [] for col in self.cols}
+        for row in all_data:
+            for col, value in zip(self.cols, row):
+                data[col].append(value)
+
+        return self.__to_readable_format(data)
+
+    def __to_readable_format(self, data):
+        """
+        Converts the data dictionary to a list for better readability.
+
+        Args:
+            data (dict): The data dictionary to convert.
+
+        Returns:
+            list: List of dictionaries with game data.
+        """
+        print(data)
+        readable_data = []
+        for i in range(len(data["Name"])):
+            game_data = {col: data[col][i] for col in self.cols}
+            readable_data.append(game_data)
+        return readable_data
```

Comment:
Have you tested the code, getting the Error - 
```bash
UnicodeEncodeError: 'charmap' codec can't encode character '\u20b9' in position 532: character maps to <undefined>
```

Also, upload Screenshot of output from the next time.

Reply:
Yes, I tested the code by scraping 300 entries for each filter.
Output Screenshot: 
<img width="460" alt="image" src="https://github.com/Clueless-Community/scrape-up/assets/75512370/f40ce8be-57b7-42fa-8701-bed2f245c695">

Also I apologies for not including it previously, as it only removed the conversion to pandas data frame and CSV. As addressed in #949